src/test-run/index.js

```diff
@@ -247,5 +249,7 @@ ServiceMessages[CLIENT_MESSAGES.ready] = function (msg) {
     if (this.lastDriverStatusResponse)
         return this.lastDriverStatusResponse;
 
-    return new Promise((resolve, reject) => this.pendingRequest = { resolve, reject });
+    var responseTimeout = setTimeout(() => this._resolvePendingRequest(null), MAX_RESPONSE_DELAY);
```

Comment:
nope. 
```js
...
this.lastDriverStatusResponse = this._handleDriverRequest(msg.status);

if (this.lastDriverStatusResponse)
    return this.lastDriverStatusResponse;

var responseTimeout = setTimeout(() => this._resolvePendingRequest(null), MAX_RESPONSE_DELAY);
...
```
If we don't have a pending command `this._handleDriverRequest(msg.status);` returns null. So, after the timeout, when we call `this._resolvePendingRequest(null)` `this.lastDriverStatusResponse` is null